Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.cpp

```diff
@@ -69,7 +69,31 @@ void CoordinatedBackingStore::paintToTextureMapper(TextureMapper& textureMapper,
     TransformationMatrix adjustedTransform = transform * TransformationMatrix::rectToRect(layerRect, targetRect);
     for (const auto& tile : m_tiles.values()) {
         ASSERT(tile.scale() == m_scale);
-        textureMapper.drawTexture(tile.texture(), tile.rect(), adjustedTransform, opacity, allTileEdgesExposed(layerRect, tile.rect()) ? TextureMapper::AllEdgesExposed::Yes : TextureMapper::AllEdgesExposed::No);
+        const auto allEdgesExposed = allTileEdgesExposed(layerRect, tile.rect()) ? TextureMapper::AllEdgesExposed::Yes : TextureMapper::AllEdgesExposed::No;
+#if ENABLE(DAMAGE_TRACKING)
+        const auto& damageRects = textureMapper.damage().rects();
+        if (!textureMapper.damage().isInvalid()
+            && !textureMapper.damage().isEmpty()
+            && adjustedTransform.isIdentity()
+            && allEdgesExposed == TextureMapper::AllEdgesExposed::No
+            && opacity == 1.0
+            && tile.texture().isOpaque()
+            && !tile.texture().filterOperation()
```

Comment:
I would move this to a helper function (or lambda) like canUseDamageToDraw() or something similar. Would it be possible in the future to support drawing texture fragment in more cases?

Reply:
I'll do it. Yes - in the future I anticipate the fragment drawing will be used in more cases.